### PR TITLE
Add `name` property for the input

### PR DESCRIPTION
**Pull request by mdluo**
In order to make use of native form input or make the typeahead able to be serialized

### CODE REVIEW

File: example/example.js [hunk 1]
```diff
@@ -95,6 +95,7 @@ const Example = React.createClass({
             align={align}
             labelKey="name"
             minLength={minLength}
+            name='typeahead'
```
Comment (ericgio) [inline example/example.js]: Can you please use double quotes here?

**Comment by mdluo** [comment]
@ericgio That's it

**Comment by mdluo** [comment]
I added one more commit to this PR @ericgio

**Comment by ericgio** [comment]
What's the use case for `defaultValue` vs. the `defaultSelected` that's already there? I'm not sure why you'd want the input to contain text but not have a selection.

**Comment by mdluo** [comment]
I'm trying to make it more close to the original `input` of React. Besides, sometimes we just need to give the input an initial string value that has nothing to do with the `option` array rather than an array containing a single object.

**Comment by ericgio** [comment]
Can you please submit the `defaultValue` change as a separate PR? I'm not convinced it's needed and I don't want that to hold up your other change.

**Comment by ericgio** [comment]
Committed #66 instead.

**Pull request closed without merging**